src/main/java/dk/kb/util/yaml/YAML.java

```diff
@@ -727,45 +776,249 @@ public static YAML resolveMultiConfig(String... configResources) throws IOExcept
+
+        if (extra instanceof LinkedHashMap) { // YAML is a LinkedHashMap
+            if (!(base instanceof LinkedHashMap)) {
+                throw new IllegalArgumentException(
+                        pre + ": Attempting to merge value type " + extra.getClass() + " to type " + base.getClass());
+            }
+            LinkedHashMap<Object, Object> bYAML = (LinkedHashMap<Object, Object>) base;
+            ((LinkedHashMap<Object, Object>)extra).forEach((key, value) -> {
+                if (!bYAML.containsKey(key)) {
+                    bYAML.put(key, value);
+                } else {
+                    switch (defaultMA) {
+                        case fail: throw new IllegalArgumentException(
+                                pre + "." + key + ": Duplicate keys with mergeAction=" + MERGE_ACTION.fail);
+                        case keep_base:
+                            break;
+                        case keep_extra:
+                            bYAML.put(key, value);
+                            break;
+                        case union:
+                            bYAML.put(key, mergeEntry(base + "." + key, bYAML.get(key), value, defaultMA, listMA));
+                            break;
+                        default: throw new UnsupportedOperationException("Unknown merge action '" + defaultMA + "'");
+                    }
+                }
+            });
+            return base;
         }
-        
-        if (!rootMap.containsKey(confRoot)) {
-            throw new NotFoundException("YAML configuration must contain the '" + confRoot + "' element", confRoot);
-        } else {
-            return rootMap.getSubMap(confRoot);
+
+        if (extra instanceof List) {
+            if (!(base instanceof List)) {
+                throw new IllegalArgumentException(
+                        pre + ": Attempting to merge value type " + List.class + " to type " + base.getClass());
+            }
+            switch (listMA) {
+                case fail: throw new IllegalArgumentException(
+                        pre + ": Duplicate keys with list merge action " + MERGE_ACTION.fail);
+                case keep_base: return base;
+                case keep_extra: return extra;
+                case union:
+                    ((List<Object>)base).addAll((List<Object>)extra);
+                    break;
+                default: throw new UnsupportedOperationException("Unknown merge action for list '" + defaultMA + "'");
+            }
+            return base;
+        }
+
+        // When it's not a map or a list we don't care about type
+        switch (defaultMA) {
+            case fail: throw new IllegalArgumentException(
+                    pre + ": Duplicate keys with merge action " + MERGE_ACTION.fail);
+            case keep_base: return base;
+            case keep_extra: return extra;
+            case union: return extra; // TODO: Should we do somthing else here? Make a type-aware merger? Fail?
```

Comment:
I think the behaviour here is quite ok, however I do believe that is should be clear from the methods Javadoc.